app/views/home/_constituency_form.html.haml

```diff
@@ -0,0 +1,14 @@
+= form_with url: pre_login_path, id: "login-form", local: true do |f|
```

Comment:
So I think here you want to do:

```
.d-flex.flex-column.justify-content-center
```

See here for example:

https://github.com/swapmyvote/swapmyvote/blob/8ca61f9a1ba59e87ee61d4e50ad55a1f28339ecf/app/views/user/swaps/_swap_profile.html.haml#L1

This should then lay out all the divs vertically and centred horizontally. You can read about the built-in BS flex classes here:

https://getbootstrap.com/docs/5.3/utilities/flex/

Reply:
Sorry, try this:
```
= form_with url: pre_login_path, id: "login-form", local: true do |f|

  .d-flex.flex-column.align-items-center

    %p.choose-constituency
      My constituency is
      = f.collection_select :constituency_ons_id, constituencies,
                            :ons_id, :name,
                            selected: default_constituency_ons_id,
                            prompt: true

    = render partial: "users/postcode_field"

    %p.text-center
      = submit_tag("Next: Choose Parties", class: "btn btn-primary")

```
The spacing still isn't right because the CSS is a bit inconsistent, this will centre it though.